gprofiler/metadata/system_metadata.py

```diff
@@ -0,0 +1,218 @@
+    # This way is used to ensure consistency
+
+    IS_64BIT = sys.maxsize > 2 ** 32
+    IFNAMSIZ = 16
+    IFF_LOOPBACK = 8
+    MAC_BYTES_LEN = 6
+    SIZE_OF_SHORT = struct.calcsize('H')
+    SIZE_OF_STUCT_ifreq = 40 if IS_64BIT else 32
+    MAX_BYTE_COUNT = 1024
+
+    buf = array.array('B', b'\0' * MAX_BYTE_COUNT)
+    ifc = struct.pack('iL', MAX_BYTE_COUNT, buf.buffer_info()[0])
+    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM, socket.IPPROTO_IP)
+    outbytes = struct.unpack('iL', fcntl.ioctl(s.fileno(), 0x8912, ifc))[0]  # SIOCGIFCONF
+    data = buf.tobytes()[:outbytes]
+    for index in range(0, len(data), SIZE_OF_STUCT_ifreq):
+        iface = data[index : index + SIZE_OF_STUCT_ifreq]
+        res = fcntl.ioctl(s.fileno(), 0x8913, iface)  # SIOCGIFFLAGS
+        ifr_flags = struct.unpack(f'{IFNAMSIZ}sH', res[: IFNAMSIZ + SIZE_OF_SHORT])[1]
+        is_loopback = ifr_flags & IFF_LOOPBACK
+        if is_loopback:
+            continue
+        res = fcntl.ioctl(s.fileno(), 0x8927, iface)  # SIOCGIFHWADDR
+        address = struct.unpack(f'{IFNAMSIZ}sH{MAC_BYTES_LEN}s', res[: IFNAMSIZ + SIZE_OF_SHORT + MAC_BYTES_LEN])[2]
+        mac = struct.unpack(f'{MAC_BYTES_LEN}B', address)
+        address = ":".join(['%02X' % i for i in mac])
+        return address
+    return "unknown"
+
+
+@dataclass
+class SystemInfo:
+    python_version: str
+    run_mode: str
+    deployment_type: str
+    kernel_release: str
+    kernel_version: str
+    system_name: str
+    processors: int
+    memory_capacity_mb: int
+    hostname: str
+    os_name: str
+    os_release: str
+    os_codename: str
+    libc_type: str
+    libc_version: str
+    hardware_type: str
+    pid: int
+    mac_address: str
+    private_ip: str
+    spawn_uptime_ms: float
+
+
+def get_static_system_info() -> SystemInfo:
+    hostname, distribution, libc_tuple, mac_address, local_ip = _initialize_system_info()
+    clock = getattr(time, "CLOCK_BOOTTIME", time.CLOCK_MONOTONIC)
+    spawn_uptime_ms = time.clock_gettime(clock)
+    libc_type, libc_version = libc_tuple
+    os_name, os_release, os_codename = distribution
+    uname = platform.uname()
+    cpu_count = os.cpu_count()
+    cpu_count = cpu_count if cpu_count is not None else 0
+    run_mode, deployment_type = get_run_mode_and_deployment_type()
+    return SystemInfo(
+        python_version=sys.version,
+        run_mode=run_mode,
+        deployment_type=deployment_type,
+        kernel_release=uname.release,
+        kernel_version=uname.version,
+        system_name=uname.system,
+        processors=cpu_count,
+        memory_capacity_mb=round(psutil.virtual_memory().total / 1024 / 1024),
+        hostname=hostname,
+        os_name=os_name,
+        os_release=os_release,
+        os_codename=os_codename,
+        libc_type=libc_type,
+        libc_version=libc_version,
+        hardware_type=uname.machine,
+        pid=os.getpid(),
+        mac_address=mac_address,
+        private_ip=local_ip,
```

Comment:
Consistent naming

Reply:
I think the better name is `local_ip`. "private" or not depends on the server

Reply:
I tend to agree, but it would require a backend change (and I'm pretty sure it's also `private_ip` in our DB so pretty hard to change). I'll change to `private_ip` for consistency, and I'll see if it's hard to change.
@leonid-granulate What do you say? Is it hard to rename `private_ip`->`local_ip`?

Reply:
It's not hard to rename, the reason we choose that names is for consistency with gAgent metadata and DB schema which is the same for both.

Reply:
@Jongy Your thoughts?

Reply:
> "private" or not depends on the server

That's what I think, lol.

Anyway, for consistency, ok, keep it.